azext_edge/edge/providers/orchestration/resources/connector/opcua/certs.py

```diff
@@ -827,6 +812,45 @@ def _begin_delete_secret(self, keyvault_name: str, secret_name: str):
                 raise
 
         # Failed to confirm deletion after retries
-        raise TimeoutError(
-            f"Failed to delete secret '{secret_name}' within {SECRET_DELETE_MAX_RETRIES} retries."
-        )
+        raise TimeoutError(f"Failed to delete secret '{secret_name}' within {SECRET_DELETE_MAX_RETRIES} retries.")
+
+    def _validate_cert_content(
+        self,
+        public_key_file: str,
+        subject_name: str,
+        application_uri: str,
+    ):
+        from cryptography import x509
+        from cryptography.x509.oid import NameOID, ExtensionOID
+
+        der_data = read_file_content(file_path=public_key_file, read_as_binary=True)
+        certificate = decode_der_certificate(der_data)
+
+        if certificate:
+            # Get the subject name and application uri from the certificate
+            # and validate it with the provided values
+            cert_subject_name = certificate.subject.get_attributes_for_oid(NameOID.COMMON_NAME)[0].value
+
+            cert_application_uri = ""
+            extension_value = certificate.extensions.get_extension_for_oid(ExtensionOID.SUBJECT_ALTERNATIVE_NAME).value
+
+            for general_name in extension_value:
+                if isinstance(general_name, x509.UniformResourceIdentifier):
+                    cert_application_uri = general_name.value
+
```

Comment:
Please remind me, what is the benefit around requiring subject name or application uri input? Is it a problem if they are always derived from the certificate?

Reply:
originally it was required since we don't process the certificate content, now that we do we can actually save 2 params for user. Will update, thanks!

Reply:
FYI: derived these 2 values from the certificate was actually the suggestion from connector team as well